subprojects/gradle/cd/src/main/kotlin/com/avito/ci/CiStepsPlugin.kt

```diff
@@ -0,0 +1,67 @@
+package com.avito.ci
+
+import com.avito.utils.gradle.BuildEnvironment
+import com.avito.utils.gradle.buildEnvironment
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.kotlin.dsl.container
+import org.gradle.kotlin.dsl.register
+
+class CiStepsPlugin : Plugin<Project> {
+
+    private val taskGroup = "ci"
+
+    override fun apply(project: Project) {
+
+        val buildContainer = project.container { name ->
+            BuildStepListExtension(name, project.objects)
+        }
+
+        project.extensions.add("builds", buildContainer)
+
+        buildContainer.registerPredefinedBuildTasks()
+
+        // TODO: MBS-6118. В CD во время конфигурации активно используется git. Это слишком долго.
+        // Локально не создаем эти таски для ускорения конфигурации
+        if (project.buildEnvironment !is BuildEnvironment.CI) return
+
+        buildContainer.all { buildTask ->
+
+            val task = project.tasks.register<Task>(buildTask.name) {
+                group = taskGroup
+                description = buildTask.description.get()
+            }
+
+            project.gradle.projectsEvaluated {
+                buildTask.steps.get().forEach { step ->
+                    step.registerTask(project, task)
+                }
+            }
+        }
+    }
+
+    @Deprecated("remove when supported in Avito")
+    private fun NamedDomainObjectContainer<BuildStepListExtension>.registerPredefinedBuildTasks() {
```

Comment:
Will remove it after two more versions of project released

Reply:
Better to add version, it will help to remove it to anybody when the version will be released

Reply:
done